test/e2e/app-dir/app-inline-css/index.test.ts

```diff
@@ -15,5 +14,44 @@ describe('app dir - css - experimental inline css', () => {
       const p = await browser.elementByCss('p')
       expect(await p.getComputedCss('color')).toBe('rgb(255, 255, 0)') // yellow
     })
+
+    it('should not return rsc payload with inlined style as a dynamic client nav', async () => {
+      let rscRequest
+
+      const browser = await next.browser('/', {
+        beforePageLoad: async (page) => {
+          page.on('request', (request) => {
+            if (request.url().includes('_rsc')) {
+              rscRequest = request
+            }
+          })
+        },
+      })
+
+      await browser.waitForElementByCss('#link-a').click()
+      await browser.waitForElementByCss('#page-a')
+
+      const res = await next.fetch(rscRequest.url(), {
+        method: 'GET',
+        headers: rscRequest.headers(),
+      })
+
+      const payload = await res.text()
```

Comment:
I'd suggest either we inspect the RSC response (via `page.on('response')`) and check that body, or skip the browser navigation all together and just fetch the RSC endpoint, since I don't see a reason to do both.